flow/Trace.h

```diff
@@ -269,11 +269,11 @@ struct TraceableString<char[S]> {
 	}
 
 	static bool atEnd(const char* value, const char* iter) {
-		return iter - value == S;
+		return iter - value == S - 1; // Exclude trailing \0 byte
```

Comment:
Are there any cases where the last byte might not be a null byte?

Reply:
only if one would explicitly pass a char[S] to detail. This doesn’t happen anywhere currently but sadly we couldn’t find a way to differentiate between literal strings and char arrays. 

But tbf: I think this problem existed before and therefore I don’t believe this would be a regression.

Reply:
We could put in an `ASSERT_WE_THINK` that the last byte is null. That way if such an event is ever logged in simulation without a terminating null byte we would discover it.

Reply:
Done